actors/builtin/miner/miner_actor.go

```diff
@@ -1777,14 +1796,32 @@ func validateFRDeclaration(deadlines *Deadlines, deadline *DeadlineInfo, declare
 // Computes a fee for a collection of sectors and unlocks it from unvested funds (for burning).
 // The fee computation is a parameter.
 func unlockPenalty(st *State, store adt.Store, currEpoch abi.ChainEpoch, sectors []*SectorOnChainInfo,
-	feeCalc func(info *SectorOnChainInfo) abi.TokenAmount) (abi.TokenAmount, error) {
+	feeCalc func(size abi.SectorSize, info *SectorOnChainInfo) abi.TokenAmount) (abi.TokenAmount, error) {
 	fee := big.Zero()
 	for _, s := range sectors {
-		fee = big.Add(fee, feeCalc(s))
+		sectorSize, err := s.Info.SealProof.SectorSize()
+		if err != nil {
+			return abi.NewTokenAmount(0), errors.Wrap(err, "could not get sector size for sector")
+		}
+		fee = big.Add(fee, feeCalc(sectorSize, s))
 	}
 	return st.UnlockUnvestedFunds(store, currEpoch, fee)
 }
 
+func computeDeclaredFaultPenalty(epochTargetReward abi.TokenAmount, networkQAPower abi.StoragePower) func(abi.SectorSize, *SectorOnChainInfo) abi.TokenAmount {
+	return func(size abi.SectorSize, s *SectorOnChainInfo) abi.TokenAmount {
+		qaPower := QAPowerForSector(size, s)
```

Comment:
I'm pretty sure penalties should be linear with power, so for multiple sectors one could sum all the power first, and then compute the penalty for that power. That might give cleaner control flow.